src/components/PopupWithForm.js

```diff
@@ -40,6 +42,16 @@ export default class PopupWithForm extends Popup {
     });
   }
 
+  setLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+      this._submitButton.disabled = true;
+    } else {
+      this._submitButton.textContent = this._submitButtonContent;
+      this._submitButton.disabled = false;
+    }
+  }
+
   close() {
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.